services/core-api/app/api/projects/project/models/project.py

```diff
@@ -76,6 +80,126 @@ def find_by_project_guid(cls, project_guid):
     @classmethod
     def find_by_mine_guid(cls, mine_guid):
         return cls.query.filter_by(mine_guid=mine_guid).all()
+    
+    @classmethod
+    def validate_project_basic_info(cls, data):
+        basic_info_schema = {
+            'project_summary_title': {
+                'type': 'string',
+                'required': True,
+            },
+            'project_summary_description': {
+                'type': 'string',
+                'required': True,
+            },
+            'proponent_project_id': {
+                'type': 'string',
+                'nullable': True,
+            }
+        }
+
+        v = Validator(basic_info_schema, purge_unknown=True)
+       
+        if not v.validate(data):
+            errors = json.dumps(v.errors)
+            current_app.logger.info(f'Validation failed for Project Basic Information with error {errors}')
+            current_app.logger.info(data)
+            raise BadRequest(errors)
+    
+    @classmethod
+    def validate_project_contacts(cls, contact):
+        base_schema = {
+            'is_primary': {
+                'type': 'boolean',
+                'required': True
+            },
+            'first_name': {
+                'type': 'string',
+                'required': True,
+            },
+            'last_name': {
+                'type': 'string',
+                'required': True,
+            },
+            'email': {
+                'type': 'string',
+                'required': True,
+            },
+            'phone_extension': {
+                'type': 'string',
+                'nullable': True,
+            },
+            'project_contact_guid' : {
+                'type': 'string',
+                'nullable': True,
+            },
+            'project_guid': {
+                'type': 'string',
+                'nullable': True,
+            },
+            'company_name': {
+                'type': 'string',
+                'nullable': True,
+            },
+            'job_title': {
+                'type': 'string',
+                'nullable': True,
+            },
+        }
+
+        contact_na_phone_schema = {
+            'phone_number': {
+                'type': 'string',
+                'required': True,
+                'regex': '[0-9]{3}-[0-9]{3}-[0-9]{4}',
+            },
+        }
+
+        contact_int_phone_schema = {
+            'phone_number': {
+                'type': 'string',
+                'required': True,
+                'maxlength': 50,
+            },
+        }     
+
+        is_primary = contact.get('is_primary')
+        address = contact.get('address', None)
+        address_type_code = address.get('address_type_code') if address != None else ''
+
```

Comment:
Thinking that maybe default for address_type_code should be "CAN"? Unless this is handled on a different level.

Reply:
Whoops lol, this is changed now.